src/poprox_recommender/training/train.py

```diff
@@ -20,90 +23,84 @@
+            output_dir=fspath(args.output_dir),
+            logging_strategy="steps",
+            save_total_limit=5,
+            lr_scheduler_type="constant",
+            weight_decay=0.0,
+            optim="adamw_torch",
+            save_strategy="epoch",
+            eval_strategy="no",
+            gradient_accumulation_steps=8,
+            learning_rate=1e-4,
+            per_device_train_batch_size=16,
+            per_device_eval_batch_size=1,
+            num_train_epochs=3,
+            remove_unused_columns=False,
+            logging_dir=fspath(args.output_dir),
+            logging_steps=500,
+            report_to=None,
+        )
 
-    # 2. Load and create datasets
-    logger.info("Initialize Dataset")
-    train_dataset = BaseDataset(
-        args,
-        root / "data/MINDlarge_post_train/behaviors_parsed.tsv",
-        root / "data/MINDlarge_post_train/news_parsed.tsv",
-    )
-    logger.info(f"The size of train_dataset is {len(train_dataset)}.")
-    eval_dataset = ValDataset(
-        args, root / "data/MINDlarge_dev/behaviors.tsv", root / "data/MINDlarge_post_dev/news_parsed.tsv"
-    )
-    logger.info(f"The size of eval_dataset is {len(eval_dataset)}.")
-
-    # 3. Train model
-    logger.info("Training Start")
-    training_args = TrainingArguments(
-        output_dir=fspath(args.output_dir),
-        logging_strategy="steps",
-        save_total_limit=5,
-        lr_scheduler_type="constant",
-        weight_decay=0.0,
-        optim="adamw_torch",
-        save_strategy="epoch",
-        eval_strategy="no",
-        gradient_accumulation_steps=8,
-        learning_rate=1e-4,
-        per_device_train_batch_size=16,
-        per_device_eval_batch_size=1,
-        num_train_epochs=3,
-        remove_unused_columns=False,
-        logging_dir=fspath(args.output_dir),
-        logging_steps=500,
-        report_to=None,
-    )
-
-    trainer = Trainer(
-        model=model,
-        args=training_args,
-        train_dataset=train_dataset,
-        eval_dataset=eval_dataset,
-    )
-
-    trainer.train()
+        trainer = Trainer(
+            model=model,
+            args=training_args,
+            train_dataset=train_dataset,
+            eval_dataset=eval_dataset,
+        )
 
-    """
-    torch.autograd.set_detect_anomaly(True)
+        # Report metrics and checkpoints to Ray Train
+        callback = ray.train.huggingface.transformers.RayTrainReportCallback()
+        trainer.add_callback(callback)
 
-    # check gradients for NaNs after training
-    def check_nan(tensor, name):
-        if torch.isnan(tensor).any():
-            raise ValueError(f"NaN in {name}: {tensor}")
+        # Prepare Transformers trainer
+        trainer = ray.train.huggingface.transformers.prepare_trainer(trainer)
 
-    for param_name, param_value in model.named_parameters():
-        if param_value.grad is not None:
-            check_nan(param_value.grad, f"gradient[{param_name}]")
-    """
+        trainer.train()
```

Comment:
If we wrap this in a `Task`, we can get subprocess task tracking too:

```python
with Task("training worker", subprocess=True) as task:
    trainer.train()
send_task(task)
```

This is necessary for the CPU time statistics to be accurate, but _not_ for the power statistics.

(The ergonomics of this could maybe use some work. Perhaps `send_task` should be automatic. But with current code, that's how you wire up a subprocess task meter.)